src/config/rootcheck-config.c

```diff
@@ -191,105 +274,187 @@ int Read_Rootcheck(XML_NODE node, void *configp, __attribute__((unused)) void *m
+                } else {
+                    snprintf(message, OS_FLSIZE,
+                        "Invalid value for element '%s': %s.",
+                        node[i]->element, node[i]->content);
+                    wm_strcat(output, message, '\n');
+                }
                 return (OS_INVALID);
             }
 #endif
         } else if (strcmp(node[i]->element, xml_check_winaudit) == 0) {
 #ifdef WIN32
             rootcheck->checks.rc_winaudit = eval_bool(node[i]->content);
             if (rootcheck->checks.rc_winaudit == OS_INVALID) {
-                merror(XML_VALUEERR, node[i]->element, node[i]->content);
+                if (output == NULL){
+                    merror(XML_VALUEERR, node[i]->element, node[i]->content);
+                } else {
+                    snprintf(message, OS_FLSIZE,
+                        "Invalid value for element '%s': %s.",
+                        node[i]->element, node[i]->content);
+                    wm_strcat(output, message, '\n');
+                }
                 return (OS_INVALID);
             }
 #endif
         } else if (strcmp(node[i]->element, xml_check_winmalware) == 0) {
 #ifdef WIN32
             rootcheck->checks.rc_winmalware = eval_bool(node[i]->content);
             if (rootcheck->checks.rc_winmalware == OS_INVALID) {
-                merror(XML_VALUEERR, node[i]->element, node[i]->content);
+                if (output == NULL){
+                    merror(XML_VALUEERR, node[i]->element, node[i]->content);
+                } else {
+                    snprintf(message, OS_FLSIZE,
+                        "Invalid value for element '%s': %s.",
+                        node[i]->element, node[i]->content);
+                    wm_strcat(output, message, '\n');
+                }
                 return (OS_INVALID);
             }
 #endif
-        } else {
+        } else if (output == NULL){
             merror(XML_INVELEM, node[i]->element);
             return (OS_INVALID);
+        } else {
+            snprintf(message, OS_FLSIZE,
+                "Invalid element in the configuration: '%s'.",
+                node[i]->element);
+            wm_strcat(output, message, '\n');
+            return (OS_INVALID);
         }
         i++;
     }
     return (0);
 }
 
-int Test_Rootcheck(const char * path){
-    int fail = 0;
+int Test_Rootcheck(const char *path, int type, char **output){
     rkconfig test_rootcheck = { .workdir = 0 };
 
-    if (ReadConfig(CAGENT_CONFIG | CROOTCHECK, path, &test_rootcheck, NULL) < 0) {
-        merror(RCONFIG_ERROR,"Rootcheck", path);
-		fail = 1;
-	}
+    if (ReadConfig(CROOTCHECK | type, path, &test_rootcheck, NULL, output) < 0) {
+        if (output == NULL){
+            merror(CONF_READ_ERROR, "Rootcheck");
+        } else {
+            wm_strcat(output, "ERROR: Invalid configuration in Rootcheck", '\n');
+        }
+	    Free_Rootcheck(&test_rootcheck);
```

Comment:
This sentence is the same as line 453. Please, modify it so that it doesn't repeat itself.